### PR TITLE
Improve docs about compatibility codes

### PR DESCRIPTION
**Pull request by Lojjs**
Documentation about 01N42, 02N42, 03N42. Also tried to remove some duplication by only referring to one single place so the content does not diverge.

### CODE REVIEW

**Comment by Lojjs** [comment]
[Add GQLStatusInfoCodes returned in case of old server-new driver (compatibility issues)](https://trello.com/c/v0bAe7Kz/450-add-gqlstatusinfocodes-returned-in-case-of-old-server-new-driver-compatibility-issues)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-status-codes-172.surge.sh

Updated pages:
- [Server notifications](https://neo4j-docs-status-codes-172.surge.sh/status-codes/5/notifications)

File: modules/ROOT/pages/notifications/index.adoc [hunk 1]
```diff
@@ -276,10 +273,36 @@ The current Notification API is still present, and the GqlStatusObject API can b
 To fully utilize the GqlStatusObject API, both your server and the driver must support it.
 Drivers that are older than 5.23 send only notifications from the Notification API, even if the server is 5.22 or later.
 
-If the 5.23 driver talks to a server that is too old to be aware of GQL-status objects, the driver tries to deduce `SUCCESS`, `OMITTED RESULT`, or `NO DATA` from the returned number of records and columns.
-If that fails, it uses the GQLSTATUS 02N42 (`no data: unknown subcondition. Unknown GQLSTATUS from old server.`) to indicate that the server did not provide a GQL-status object.
-Then, it poly-fills statuses from the notifications and merges the two lists according to the GQL precedence rules described in <<gqlstatus-notification-object, GQL-status notification object>>.
+If a driver of version 5.23 or later talks to a server that is too old to be aware of GQL-status objects, the driver needs to poly-fill the GqlStatusObject API with information.
+The driver tries to deduce `SUCCESS`, `OMITTED RESULT`, or `NO DATA` from the returned number of records and columns.
+If that fails, the general GQLSTATUS code will be set to `02N42` .
```
Comment (NataliaIvakina) [inline modules/ROOT/pages/notifications/index.adoc:278]:
```suggestion
If that fails, the general GQLSTATUS code will be set to `02N42`.
```

File: modules/ROOT/pages/notifications/index.adoc [hunk 2]
```diff
@@ -276,10 +273,36 @@ The current Notification API is still present, and the GqlStatusObject API can b
...
+Then, the driver poly-fills the rest of the GQL-status object list with the notifications from the old notification API.
+These will get GQLSTATUS `01N42` for notifictions with severity `WARNING` and `03N42` for notifications with severity `INFORMATION`.
```
Comment (NataliaIvakina) [inline modules/ROOT/pages/notifications/index.adoc:280]:
```suggestion
These will get GQLSTATUS `01N42` for notifications with severity `WARNING` and `03N42` for notifications with severity `INFORMATION`.
```

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-status-codes-172.surge.sh

Updated pages:
- [Server notifications](https://neo4j-docs-status-codes-172.surge.sh/status-codes/5/notifications)

**Pull request merged**